admin/authentication/dfe-signon-strategy.js

```diff
@@ -0,0 +1,42 @@
+'use strict'
+
+const config = require('../config')
+const { Strategy, Issuer } = require('openid-client')
+const logger = require('../services/log.service').getLogger()
+const asyncRetry = require('login.dfe.async-retry')
+
+const initSignOn = async () => {
+  Issuer.defaultHttpOptions = { timeout: 10000 }
+  const issuer = await asyncRetry(async () => Issuer.discover(config.Auth.dfeSignIn.authUrl), asyncRetry.strategies.apiStrategy)
+  logger.debug('dfe.issuer', issuer)
+  const client = new issuer.Client({
+    client_id: config.Auth.dfeSignIn.clientId,
+    client_secret: config.Auth.dfeSignIn.clientSecret
+  })
+  if (config.Auth.dfeSignIn.clockTolerance && config.Auth.dfeSignIn.clockTolerance > 0) {
+    client.CLOCK_TOLERANCE = config.Auth.dfeSignIn.clockTolerance
+  }
+
+  return new Strategy({
+    client,
+    params: {
+      // redirect_uri: `http://localhost:3001/auth`,
+      scope: 'openid profile email'
```

Comment:
magic string?

Reply:
taken verbatim from provided example.  will figure out what this is supposed to be and action appropriately

Reply:
@jon-shipley these are the 'sets' of data you want included in the payload returned from the auth provider.  such as their email, org info etc.

This has been added as config with a default.